modules/ting_search_ortograf/includes/ting_search_ortograf.ajax.inc

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * @file
+ * AJAX functions bibliotek.dk autocomplete.
+ */
+
+/**
+ * Auto-complete callback for the ting search block.
+ *
+ * The function returns JSON and ending execution.
+ *
+ * @param string $string
+ *   The string to search suggestion with.
+ */
+function ting_search_ortograf_callback($string = '') {
+  $string = check_plain(strtolower($string));
+  $query = ting_search_ortograf_settings();
+  $minimum_string = (int) $query['minimumString'];
+  if (drupal_strlen($string) < $minimum_string) {
+    drupal_json_output(array());
+    drupal_exit();
+  }
+  // MinimumString is not a suggestion parameter.
+  unset($query['minimumString']);
+  $query['q'] = $string;
+
+  // Try to use cache for the search string.
+  $cid = md5(serialize($query));
+  if ($cache = cache_get($cid, 'cache_ting_search_ortograf')) {
+    $matches = $cache->data;
+  }
+  else {
+    $matches = ting_search_ortograf_get_suggestions($query);
+    // Store the matches in cache for faster lookup next time.
+    cache_set($cid, $matches, 'cache_ting_search_ortograf', CACHE_TEMPORARY);
+  }
+
+  drupal_json_output($matches);
+  drupal_exit();
+}
+
+
+/**
+ * SuggestionService - retrieve suggestions with drupal_http_request() 
+ *
+ * @param array $query
+ *   Search suggestion parameters.
+ *
+ * @return array
+ *   Search suggestion results.
+ */
+function ting_search_ortograf_get_suggestions(array $query) {
+  $items = array();
+  $rest_url = variable_get('ting_search_ortograf_suggestion_url', '');
+  $url = url($rest_url, array('query' => $query));
+  $headers = array('Accept' => 'application/json');
+
+  $result = drupal_http_request($url, $headers);
+  if (!empty($result->data)) {
+    $result_data = drupal_json_decode($result->data);
+    if (empty($result->error)) {
+      if (!empty($result_data['response']['docs'])) {
+        foreach ($result_data['response']['docs'] as $suggestion) {
+          $suggestion = check_plain($suggestion['all'][0]);
+          $val = truncate_utf8($suggestion, 62, TRUE, TRUE, 1);
+          $key = truncate_utf8($suggestion, 256, TRUE, FALSE, 1);
```

Comment:
This handling of the suggestions seems very specific to the autocomplete case. I suggest moving that to `ting_search_ortograf_callback()` - preferably with a comment describing why it is needed.

This way we can keep `ting_search_ortograf_get_suggestions` clean and reusable for other cases.